test/transaction/deferred_actions_test.cpp

```diff
@@ -0,0 +1,299 @@
+#include <vector>
+#include "storage/garbage_collector.h"
+#include "storage/sql_table.h"
+#include "storage/storage_defs.h"
+#include "transaction/transaction_context.h"
+#include "transaction/transaction_defs.h"
+#include "transaction/transaction_manager.h"
+#include "util/test_harness.h"
+#include "util/transaction_test_util.h"
+
+namespace terrier {
+
+class DeferredActionsTest : public TerrierTest {
+ protected:
+  DeferredActionsTest() : gc_(&txn_mgr_) {}
+
+  void SetUp() override { TerrierTest::SetUp(); }
+
+  void TearDown() override {
+    gc_.PerformGarbageCollection();
+    gc_.PerformGarbageCollection();
+    TerrierTest::TearDown();
+  }
+
+  storage::RecordBufferSegmentPool buffer_pool_ = {100, 100};
+  transaction::TransactionManager txn_mgr_ = {&buffer_pool_, true, LOGGING_DISABLED};
+  storage::GarbageCollector gc_;
+};
+
+// NOLINTNEXTLINE
+TEST_F(DeferredActionsTest, AbortAction) {
```

Comment:
nit: We usually have a comment block at the beginning of each test explaining what it does at a high level.

Reply:
I've added comments.  Please let me know if you think I need to add more depth on any of them.